libdnf/module/module_sack.cpp

```diff
@@ -478,6 +482,149 @@ std::pair<std::vector<std::vector<std::string>>, ModuleSack::ModuleErrorType> Mo
+    if (!config_platform_option.empty()) {
+        auto & config_platform = config_platform_option.get_value();
+        auto parsed_platform_id = parse_platform_id_from_string(config_platform);
+        if (!parsed_platform_id.first.empty() && !parsed_platform_id.second.empty()) {
+            return parsed_platform_id;
+        } else {
+            base->get_logger()->warning("Invalid format of platform ID in configuration: {}", config_platform);
+        }
+    }
+
+    libdnf::rpm::PackageQuery base_query(base);
+    base_query.filter_provides({"system-release"});
+    base_query.filter_latest_evr();
+
+    // try to detect platform id from available packages
+    libdnf::rpm::PackageQuery available_query(base_query);
+    available_query.filter_available();
+    auto available_provides = get_strings_from_provide(available_query, "base-module");
+    if (available_provides.size() == 1) {
+        auto & query_platform_id = *available_provides.begin();
+        auto parsed_platform_id = parse_platform_id_from_string(query_platform_id);
+        if (!parsed_platform_id.first.empty() && !parsed_platform_id.second.empty()) {
+            return parsed_platform_id;
+        } else {
+            base->get_logger()->warning(
+                "Invalid format of module platform queried from available packages: {}", query_platform_id);
+        }
+    } else {
+        base->get_logger()->warning("Multiple module platforms provided by available packages");
+    }
+
+    // try to detect platform id from installed packages
+    libdnf::rpm::PackageQuery installed_query(base_query);
+    installed_query.filter_installed();
+    auto installed_provides = get_strings_from_provide(installed_query, "base-module");
+    if (installed_provides.size() == 1) {
+        auto & query_platform_id = *installed_provides.begin();
+        auto parsed_platform_id = parse_platform_id_from_string(query_platform_id);
+        if (!parsed_platform_id.first.empty() && !parsed_platform_id.second.empty()) {
+            return parsed_platform_id;
+        } else {
+            base->get_logger()->warning(
+                "Invalid format of module platform queried from installed packages: {}", query_platform_id);
+        }
+    } else {
+        base->get_logger()->warning("Multiple module platforms provided by installed packages");
+    }
+
+    // try to detect platform id from release files
+    // TODO(jkolarik): Create constants for paths
+    std::vector<std::string> os_release_paths{"etc/os-release", "usr/lib/os-release"};
+    auto & installroot = base->get_config().installroot().get_value();
+    for (auto & os_release_path : os_release_paths) {
+        std::string full_path = std::filesystem::canonical(installroot) / os_release_path;
+        std::pair<std::string, std::string> parsed_platform_id;
+
+        try {
+            parsed_platform_id = parse_platform_id_from_file(full_path);
+        } catch (const std::exception & except) {
```

Comment:
Very minor: The catch is quite wide . I believe that only two exception are expected.

Reply:
Agree, better to catch only the things we know about and don't suppress any other potential problems.